[2026.3] Add Sure Petcare entities details and pet select location blueprint (#43822)
home-assistant/home-assistant.github.io: source/_integrations/surepetcare.markdown

{% include integrations/config_flow.md %}

## Entities

These are the entities available in the Sure Petcare integration.

### Pets

| Domain        | Name                     | Description
| ------------- | ------------------------ | ------------------------------------------------------------
| Binary sensor | Presence                 | Whether the pet is inside (presence).
| Sensor        | Last seen flap device ID | Diagnostic: last flap device ID used by the pet (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a flap update.

> | Sensor        | Last seen user ID        | Diagnostic: last user ID that manually changed pet location (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a manual update.

Reviewer: ```suggestion
#### Binary sensors

- **Presence**: Whether the pet is inside (presence).

#### Sensors

- **Last seen flap device ID**: Diagnostic: last flap device ID used by the pet (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a flap update.
- **Last seen user ID**: Diagnostic: last user ID that manually changed pet location (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a manual update.
```
We [prefer lists over tables](https://developers.home-assistant.io/docs/documenting/general-style-guide#tables), see [example](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L105).

### Pet and cat flaps

| Domain         | Name           | Description                                                                    |
| -------------- | -------------- | ------------------------------------------------------------------------------ |
| Binary sensor  | Connectivity   | Device connectivity (online); shows device RSSI when available.                |
| Lock           | Locked in      | Lock state: flap locked to allow entry only.                                   |
| Lock           | Locked out     | Lock state: flap locked to allow exit only.                                    |
| Lock           | Locked all     | Lock state: flap locked both ways.                                             |

> | Sensor         | Battery level  | Battery level percentage (derived from battery voltage).                       |

Reviewer: ```suggestion
#### Binary sensors

- **Connectivity**: Device connectivity (online); shows device RSSI when available.

#### Locks

- **Locked in**: Lock state: flap locked to allow entry only.
- **Locked out**: Lock state: flap locked to allow exit only.
- **Locked all**: Lock state: flap locked both ways.

#### Sensors

- **Battery level**: Battery level percentage (derived from battery voltage).
```

### Feeders

| Domain         | Name           | Description                                                                    |
| -------------- | -------------- | ------------------------------------------------------------------------------ |
| Binary sensor  | Connectivity   | Feeder connectivity (online); shows device RSSI when available.                |

> | Sensor         | Battery level  | Battery level percentage (derived from battery voltage).                       |

Reviewer: ```suggestion
#### Binary sensors

- **Connectivity**: Feeder connectivity (online); shows device RSSI when available.

#### Sensors

- **Battery level**: Battery level percentage (derived from battery voltage).            |
```

### Felaqua

| Domain | Name    | Description                  |
| ------ | ------- | ---------------------------- |

> | Sensor | Felaqua | Water remaining in the bowl. |

Reviewer: ```suggestion
#### Sensors

- **Felaqua**: Water remaining in the bowl.
```

### Hub

| Domain         | Name           | Description                                                                  |
| -------------- | -------------- | ---------------------------------------------------------------------------- |

> | Binary sensor  | Connectivity   | Hub connectivity (online); attributes include `led_mode` and `pairing_mode`. |

Reviewer: ```suggestion
#### Binary sensors

- **Connectivity**: Hub connectivity (online); attributes include `led_mode` and `pairing_mode`.
```

{% include integrations/config_flow.md %}

> ## Entities

Reviewer: ```suggestion
## Supported functionality
```

These are the entities available in the Sure Petcare integration.

### Pets

#### Binary sensors

- **Presence**: Whether the pet is inside (presence).

#### Sensors

- **Last seen flap device ID**: Diagnostic: last flap device ID used by the pet (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a flap update.

> - **Last seen user ID**: Diagnostic: last user ID that manually changed pet location (useful for the [pet select location blueprint](#pet-select-location-template-entity)); `unknown` if the last status is not from a manual update.

Reviewer: ```suggestion
- **Last seen flap device ID**: Last flap device ID used by the pet (useful for the [pet select location blueprint](#pet-select-location-template-entity)). `unknown` if the last status is not from a flap update.
- **Last seen user ID**: Last user ID that manually changed the pet location (useful for the [pet select location blueprint](#pet-select-location-template-entity)). `unknown` if the last status is not from a manual update.
```

### Pet and cat flaps

#### Binary sensors

- **Connectivity**: Device connectivity (online); shows device RSSI when available.

#### Locks

- **Locked in**: Lock state: flap locked to allow entry only.
- **Locked out**: Lock state: flap locked to allow exit only.
- **Locked all**: Lock state: flap locked both ways.

#### Sensors

- **Battery level**: Battery level percentage (derived from battery voltage).

### Feeders

#### Binary sensors

- **Connectivity**: Feeder connectivity (online); shows device RSSI when available.

#### Sensors

- **Battery level**: Battery level percentage (derived from battery voltage).            |

### Felaqua

#### Sensors

> - **Felaqua**: Water remaining in the bowl.

Reviewer: Is this sensor really called "Felaqua"? Not "Water remaining" or something along those lines?

Author: I don't have any Felaqua myself to test it but looking at the code, the entity should have the same name as the device if I am not mistaken:

(from [here](https://github.com/home-assistant/core/blob/d6f355355fe2d8ecae9070810b0fe5d104c7e0e0/homeassistant/components/surepetcare/sensor.py#L106))
```python
self._attr_name = self._device_name
self._attr_unique_id = self._device_id
self._attr_entity_picture = surepy_entity.icon
```

> - **Felaqua**: Water remaining in the bowl.

Reviewer: ```suggestion
- **Battery level**: Battery level percentage (derived from battery voltage).
- **Felaqua**: Water remaining in the bowl, in milliliters.
```
is there also a battery level for Felaqua?

Author: Yes, I also forgot to add the connectivity binary sensor to it. I will add those.
